mlflow/entities/conversion_utils.py

```diff
@@ -0,0 +1,117 @@
+try:
+    import numpy as np
+    __import_np = True
+except ImportError:
+    __import_np = False
+
+try:
+    import pyspark.ml.param
+    __import_pyspark_ml_param = True
+except ImportError:
+    __import_pyspark_ml_param = False
+
+
+try:
+    import tensorflow as tf
+    __import_np = True
+except ImportError:
+    __import_tf = False
+
+try:
+    import torch
+    __import_torch = True
```

Comment:
Can we avoid importing pyspark/torch/tensorflow when we importing this `conversion_utils` module ? We can move these lines into function.

Reply:
You don't need to use flag like `__np_module_imported`

if we execute `import xxx` command twice, the second execution do nothing, because when we import one module, it is cached in `sys.modules`.